workspaces/local-cli/src/commands/status.ts

```diff
@@ -0,0 +1,364 @@
+  async getDiffsAndEvents(
+    paths: IPathMapping,
+    captureId: string,
+  ) {
+    const daemonState = await ensureDaemonStarted(
+      lockFilePath,
+      Config.apiBaseUrl
+    );
+
+    const apiBaseUrl = `http://localhost:${daemonState.port}/api`;
+    developerDebugLogger(`api base url: ${apiBaseUrl}`);
+    const cliClient = new Client(apiBaseUrl);
+    cliClient.setIdentity(await getUser());
+    const cliSession = await cliClient.findSession(paths.cwd, null, null);
+    developerDebugLogger({ cliSession });
+
+    const eventEmitter = new EventEmitter();
+    const specService = new SpecServiceClient(
+      cliSession.session.id,
+      eventEmitter,
+      apiBaseUrl
+    );
+
+    const ignoreHelper = new IgnoreFileHelper(
+      paths.opticIgnorePath,
+      paths.configPath
+    );
+    const ignoreRules = (await ignoreHelper.getCurrentIgnoreRules()).allRules;
+    const events = await fs.readJSON(paths.specStorePath);
+
+    const startDiffUrl = `${apiBaseUrl}/specs/${cliSession.session.id}/captures/${captureId}/diffs`;
+
+    const { diffId, notificationsUrl } = await JsonHttpClient.postJson(
+      startDiffUrl,
+      {
+        ignoreRules,
+        additionalCommands: [],
+        events,
+        filters: [],
+      }
+    );
+
+    const getDiffsUrl = `${apiBaseUrl}/specs/${cliSession.session.id}/captures/${captureId}/diffs/${diffId}/diffs`;
+    const getUndocumentedUrlsUrl = `${apiBaseUrl}/specs/${cliSession.session.id}/captures/${captureId}/diffs/${diffId}/undocumented-urls`;
+
+    const notificationChannel = new EventSource(
+      `http://localhost:${daemonState.port}` + notificationsUrl
+    );
+
+    // let customBar: any | undefined;
+    let totalInteractions = 0
+    cli.action.start('computing diffs for observations')
+    specService.getCaptureStatus(captureId).then(({ interactionsCount }) => {
+      totalInteractions = interactionsCount
+    });
+
+    await new Promise((resolve, reject) => {
+      notificationChannel.onmessage = (event) => {
+        const { type, data } = JSON.parse(event.data);
+        if (type === 'message') {
+          cli.action.start(
+            `computing diffs for observations ${parseInt(data.diffedInteractionsCounter) +parseInt(data.skippedInteractionsCounter)}${totalInteractions > 0 ? `/${totalInteractions.toString()}`: ''}`)
+          if (!data.hasMoreInteractions) {
+            cli.action.stop('done')
+            resolve();
+          }
+        } else if (type === 'error') {
+          cli.action.stop('done')
+          reject();
+        }
+      };
+      notificationChannel.onerror = function (err) {
+        reject();
+      };
+    });
+
+    notificationChannel.close();
+
+    const diffs = await JsonHttpClient.getJson(getDiffsUrl);
+    const undocumentedUrls = await JsonHttpClient.getJson(
+      getUndocumentedUrlsUrl
+    );
+
+    return {
+      diffs,
+      events,
+      undocumentedUrls: undocumentedUrls.urls,
+    };
+  }
+
+  private printStatus(
```

Comment:
this would be more testable if it returned strings or an iterator of strings as opposed to imperatively using this.log()